src/repository/NodeRepository.ts

```diff
@@ -42,15 +50,61 @@ export class NodeRepository {
         return this.connectionPool.queryPaginated<NodeRow[]>(sql, params)
     }
 
+    async getNeighbors(
+        nodeId?: DhtAddress,
+        streamPartId?: StreamPartID,
+        pageSize?: number,
+        cursor?: string
+    ): Promise<PaginatedListFragment<NeighborRow[]>> {
+        logger.info('Query: getNeighbors', { nodeId, streamPartId })
+        const whereClauses = []
+        const params = []
+        if (nodeId !== undefined) {
+            whereClauses.push('nodeId1 = ? OR nodeId2 = ?')
+            params.push(nodeId, nodeId)
+        }
+        if (streamPartId !== undefined) {
+            whereClauses.push('streamPartId = ?')
+            params.push(streamPartId)
+        }
+        const sql = createSqlQuery(
+            'SELECT streamPartId, nodeId1, nodeId2 FROM neighbors',
+            whereClauses
+        )
+        return this.connectionPool.queryPaginated<NeighborRow[]>(
+            sql,
+            params,
+            pageSize,
+            cursor
+        )
+    }
+
     async replaceNetworkTopology(topology: Topology): Promise<void> {
         const nodes = topology.getNodes().map((node) => {
             return [node.id, node.ipAddress]
         })
+        const neighbors: [StreamPartID, DhtAddress, DhtAddress][] = []
+        for (const node of topology.getNodes()) {
+            for (const streamPartId of node.streamPartNeighbors.keys()) {
+                const streamPartNeighbors = node.streamPartNeighbors.get(streamPartId)!
```

Comment:
Matter of taste of course, but what about using `.entries` to avoid extra getter call?
```suggestion
            for (const [streamPartId, streamPartNeighbors] of node.streamPartNeighbors.entries()) {
```